src/sentry/rules/processor.py

```diff
@@ -303,9 +322,26 @@ def activate_downstream_actions(self, rule: Rule) -> None:
                 continue
 
             action_inst = action_cls(self.project, data=action, rule=rule)
-            results = safe_execute(
-                action_inst.after, event=self.event, state=state, _with_transaction=False
-            )
+
+            #  notification_uuid is needed for email, Discord, Slack, and MS Teams
+            action_with_notification_uuid = [
+                NotifyEmailAction,
+                DiscordNotifyServiceAction,
+                SlackNotifyServiceAction,
+                MsTeamsNotifyServiceAction,
+            ]
+            if action_cls in action_with_notification_uuid:
+                results = safe_execute(
+                    action_inst.after,
+                    event=self.event,
+                    state=state,
+                    _with_transaction=False,
+                    notification_uuid=notification_uuid,
+                )
+            else:
+                results = safe_execute(
+                    action_inst.after, event=self.event, state=state, _with_transaction=False
+                )
```

Comment:
Should we just assume all actions are passed a notification_uuid, and have that part of the signature for all actions?

Reply:
We could, but it might make the code a bit confusing, as we'll only be using the notification_uuid in some of the `after` functions. Is there a reason this would be preferred to `**kwargs`?

Reply:
Thank you - that makes sense, I'll make the change